### PR TITLE
[TSI] Add CL implementation for Create and Delete Types

### PR DESCRIPTION
**Pull request by bikamani**
I will have follow up PR to add tests.

### CODE REVIEW

File: sdk/timeseriesinsights/Azure.Iot.TimeSeriesInsights/src/TimeSeriesInsightsClient.cs [hunk 1]
```diff
@@ -1380,5 +1380,315 @@ Page<TimeSeriesType> NextPageFunc(string nextLink, int? pageSizeHint)
...
+        /// <exception cref="ArgumentException">
+        /// The exception is thrown when <paramref name="timeSeriesTypeIds"/> is empty.
+        /// </exception>
+        public virtual Response<TimeSeriesOperationError[]> DeleteTimeSeriesTypesbyId(
+            IEnumerable<string> timeSeriesTypeIds,
+            CancellationToken cancellationToken = default)
+        {
+            using DiagnosticScope scope = _clientDiagnostics.CreateScope($"{nameof(TimeSeriesInsightsClient)}.{nameof(DeleteInstances)}");
+            scope.Start();
+            // 2, 1, 3
```
Comment (bikamani) [inline sdk/timeseriesinsights/Azure.Iot.TimeSeriesInsights/src/TimeSeriesInsightsClient.cs:1669]: deleting this, left from testing.

File: sdk/timeseriesinsights/Azure.Iot.TimeSeriesInsights/src/TimeSeriesInsightsClient.cs [hunk 2]
```diff
@@ -1380,5 +1380,314 @@ Page<TimeSeriesType> NextPageFunc(string nextLink, int? pageSizeHint)
...
+        {
+            using DiagnosticScope scope = _clientDiagnostics
+                .CreateScope($"{nameof(TimeSeriesInsightsClient)}.{nameof(CreateOrReplaceTimeSeriesTypes)}");
+            scope.Start();
+
+            try
+            {
+                Argument.AssertNotNullOrEmpty(timeSeriesTypes, nameof(timeSeriesTypes));
+
+                TypesBatchRequest batchRequest = new TypesBatchRequest();
```
Comment (drwill-ms) [inline sdk/timeseriesinsights/Azure.Iot.TimeSeriesInsights/src/TimeSeriesInsightsClient.cs:1414]: var

File: sdk/timeseriesinsights/Azure.Iot.TimeSeriesInsights/src/TimeSeriesInsightsClient.cs [hunk 3]
```diff
@@ -1380,5 +1380,314 @@ Page<TimeSeriesType> NextPageFunc(string nextLink, int? pageSizeHint)
...
+        /// <remarks>
+        /// For more samples, see <see href="https://github.com/Azure/azure-sdk-for-net/tree/master/sdk/timeseriesinsights/Azure.Iot.TimeSeriesInsights/samples">our repo samples</see>.
+        /// </remarks>
+        /// <exception cref="ArgumentNullException">
+        /// The exception is thrown when <paramref name="timeSeriesTypes"/> is <c>null</c>.
+        /// </exception>
+        /// <exception cref="ArgumentException">
+        /// The exception is thrown when <paramref name="timeSeriesTypes"/> is empty.
+        /// </exception>
+        public virtual async Task<Response<TimeSeriesOperationError[]>> CreateOrReplaceTimeSeriesTypesAsync(
```
Comment (drwill-ms) [inline sdk/timeseriesinsights/Azure.Iot.TimeSeriesInsights/src/TimeSeriesInsightsClient.cs:1402]: Can it really be "OrReplace" if there is no way to pass an ETag?

**Review by timtay-microsoft — APPROVED** [review]

File: sdk/timeseriesinsights/Azure.Iot.TimeSeriesInsights/src/TimeSeriesInsightsClient.cs (hunk 3 above)
Comment (timtay-microsoft) [reply sdk/timeseriesinsights/Azure.Iot.TimeSeriesInsights/src/TimeSeriesInsightsClient.cs:1402]: From what I recall, it does still replace. Without the etag, there is just no concurrency protection around it

File: sdk/timeseriesinsights/Azure.Iot.TimeSeriesInsights/src/TimeSeriesInsightsClient.cs [hunk 4]
```diff
@@ -1380,5 +1380,314 @@ Page<TimeSeriesType> NextPageFunc(string nextLink, int? pageSizeHint)
...
+            CancellationToken cancellationToken = default)
+        {
+            using DiagnosticScope scope = _clientDiagnostics.CreateScope($"{nameof(TimeSeriesInsightsClient)}.{nameof(DeleteTimeSeriesTypesByNames)}");
+            scope.Start();
+
+            try
+            {
+                Argument.AssertNotNullOrEmpty(timeSeriesTypeNames, nameof(timeSeriesTypeNames));
+
+                TypesBatchRequest batchRequest = new TypesBatchRequest()
```
Comment (drwill-ms) [inline sdk/timeseriesinsights/Azure.Iot.TimeSeriesInsights/src/TimeSeriesInsightsClient.cs:1519]: No () needed when no ctor params and using initializer list

**Review by drwill-ms — APPROVED** [review]

File: sdk/timeseriesinsights/Azure.Iot.TimeSeriesInsights/src/TimeSeriesInsightsClient.cs (hunk 3 above)
Comment (bikamani) [reply sdk/timeseriesinsights/Azure.Iot.TimeSeriesInsights/src/TimeSeriesInsightsClient.cs:1402]: yes if anything changes on the type parameter, it does replace.

File: sdk/timeseriesinsights/Azure.Iot.TimeSeriesInsights/src/TimeSeriesInsightsClient.cs (hunk 2 above)
Comment (bikamani) [reply sdk/timeseriesinsights/Azure.Iot.TimeSeriesInsights/src/TimeSeriesInsightsClient.cs:1414]: changed here and other places below

**Pull request merged**